### PR TITLE
Fix public chat join request miss invite link

### PR DESCRIPTION
**Pull request by omg-xtao**
public chat join request will raise `invite` miss `admin_id` attr

fixes: #1037

### CODE REVIEW

File: pyrogram/types/user_and_chats/chat_invite_link.py [hunk 1]
```diff
@@ -104,11 +104,12 @@ def _parse(
...
     ) -> "ChatInviteLink":
-        creator = (
-            types.User._parse(client, users[invite.admin_id])
-            if users is not None
-            else None
-        )
+        if users is not None:
+            if not hasattr(invite, "admin_id"):
+                return None
+            creator = types.User._parse(client, users[invite.admin_id])
```
Comment (AndrielFR) [inline pyrogram/types/user_and_chats/chat_invite_link.py:110]:
```python
        creator = (
            types.User._parse(client, users[invite.admin_id])
            if users is not None and hasattr(invite, "admin_id")
            else None
        )
```
is better

**Review by AndrielFR — CHANGES_REQUESTED** [review]

File: pyrogram/types/user_and_chats/chat_invite_link.py (hunk 1 above)
Comment (omg-xtao) [reply pyrogram/types/user_and_chats/chat_invite_link.py:110]: This shouldn't be the case because the invite instance is empty.

Comment (test123214142343242) [reply pyrogram/types/user_and_chats/chat_invite_link.py:110]:
```if not hasattr(invite, "admin_id"):
    return None
if users is not None:
    creator = types.User._parse(client, users[invite.admin_id])
else:
    creator = None```

Comment (test123214142343242) [reply pyrogram/types/user_and_chats/chat_invite_link.py:110]: yes 

'ChatInvitePublicJoinRequests' object has no attribute 'link'
Traceback (most recent call last):
  File "venv/lib/python3.10/site-packages/pyrogram/dispatcher.py", line 214, in handler_worker
    await parser(update, users, chats)
  File "venv/lib/python3.10/site-packages/pyrogram/dispatcher.py", line 126, in chat_join_request_parser
    pyrogram.types.ChatJoinRequest._parse(self.client, update, users, chats),
  File "venv/lib/python3.10/site-packages/pyrogram/types/user_and_chats/chat_join_request.py", line 81, in _parse
    invite_link=types.ChatInviteLink._parse(client, update.invite, users),
  File "venv/lib/python3.10/site-packages/pyrogram/types/user_and_chats/chat_invite_link.py", line 116, in _parse
    invite_link=invite.link,
AttributeError: 'ChatInvitePublicJoinRequests' object has no attribute 'link'

**Pull request closed without merging**